controllers/dexclient_controller.go

```diff
@@ -106,7 +111,7 @@ func (r *DexClientReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
 			dexv1Client.Spec.ClientID, // name
 			dexv1Client.Name,          // clientid
 			dexv1Client.Spec.LogoURL,
-			dexv1Client.Spec.ClientSecret, // secret
+			dexclientclientSecret, // secret
```

Comment:
Do we need the same change for UpdateClient? Will this handle the case where the secret changes?

Reply:
You will need to add a watcher on the secret in order to regenerate the dex server configuration.

Reply:
we only set the dexclient secret on create
we don't update the secret, we can update all other bits but not the secret